### ISSUE TITLE
MultiFileChooser issue

### ISSUE THREAD

**Issue by rockeysc** [chriskiehl/Gooey#738]
Hi there, 

I'm new to Gooey and I'm trying to implement it into my code. Right now I'm using it with pandas and I can't get the MultiFileChooser to work. I want to no declare a specific file path and just be able to select the files then run the code. 
 
Here is my code: 

```
@Gooey(dump_build_config=True, program_name="Widget Demo")
def main():
    desc = "Example application to show Gooey's various widgets"
    file_help_msg = "Name of the file you want to process"

    my_cool_parser = GooeyParser(description=desc)

    my_cool_parser.add_argument(
        "MultiFileChooser", nargs="*", help=file_help_msg, widget="MultiFileChooser"
    )

    li = []

    df = pd.read_csv(
        index_col=None,
        header=None,
        error_bad_lines=False,
        sep=",",
        warn_bad_lines=False,
    )
    df.columns = ["ID", "Status", "Time", "Date", "Call Letters", " ", " "]
    li.append(df)

    frame = pd.concat(li, axis=0, ignore_index=True)

    frame.drop_duplicates(subset=None, inplace=True)

    list_drop = ["ID", "Status", "Time", "Date", " ", " "]
    frame.drop(list_drop, axis=1, inplace=True)

    frame.drop_duplicates(subset=None, inplace=True)
    cl = frame.transpose()

    cl.to_csv("Extraction.txt", header=None, index=False)

    my_cool_parser.add_argument("directory", help="Directory to store output")
    my_cool_parser.parse_args()
    display_message()


def here_is_more():
    pass

if __name__ == "__main__":
    main()

```

When I run it, I get the following error `TypeError: parser_f() missing 1 required positional argument: 'filepath_or_buffer'` 

So in short, how do I get the MultifileChooser to work. 

Thanks in advance!

**Comment by chriskiehl** [comment]
Your error is coming from the `read_csv` line, not from Gooey. You're not passing it any file path or buffer. 

You can verify this by just stripping everything in your code sample away except for the `read_csv` part: 

```python
df = pd.read_csv(
        index_col=None,
        header=None,
        error_bad_lines=False,
        sep=",",
        warn_bad_lines=False,
    )
```

Run that file and you'll indeed see your error: 

```
TypeError: read_csv() missing 1 required positional argument: 'filepath_or_buffer'
```

`read_csv` needs a positional argument supplied (e.g. `pd.read_csv('path/to/my/file.csv')`)

**Comment by rockeysc** [comment]
Thanks Chriskiehl, 

I guess what I need to do is figure out how to pass the file choose as my path because when I set a path in my code, the file chooser ignores whatever I choose and just looks at where the path is directed too.

**Comment by chriskiehl** [comment]
I'm not sure how you plan to use the multiple directories. However, as a small single file example to maybe point you in the right direction: everything from gooey land is available after you call `parse_args()` (hand waving away a few details here). Which means all of your logic needs to be _after_ that call in order to use any of the user's values in your code. 

So, let's say you wanted to let the user set the output file location, we can make a few modifications to your code to get it all working: 

```python
@Gooey(dump_build_config=True, program_name="Widget Demo")
def main():
    my_cool_parser = GooeyParser(description=desc)

    my_cool_parser.add_argument('output', help='choose an output file!', widget="FileSaver")
    args = parse_args()

    df = pd.read_csv('some/arbitrary/file.csv') 
    # so some work here 

    # now save the output using their selected location 
    df.to_csv(args.output, header=None, index=False)  # <-- note that the user's output path is supplied here!
```

**Comment by rockeysc** [comment]
I'm actually looking to do the opposite, I'm trying to select multiple files, do some work then save it in one file. I'm just having trouble with the parser I think. Here is my latest attempt: 

```
@Gooey(program_name="test")
def parse_args():
    stored_args = {}
    script_name = os.path.splitext(os.path.basename(__file__))[0]
    args_file = "{}-args.json".format(script_name)

    if os.path.isfile(args_file):
        with open(args_file) as data_file:
            stored_args = json.load(data_file)
    parser = GooeyParser(description="test")
    parser.add_argument(
        "data_directory",
        action="store",
        default=stored_args.get("data_directory"),
        widget="MultiFileChooser",
        help="Source directory",
    )
    parser.add_argument(
        "output_directory",
        action="store",
        widget="FileSaver",
        default=stored_args.get("output_directory"),
        help="Output directory",
    )

    args = parser.parse_args()
    with open(args_file, "w") as data_file:
        json.dump(vars(args), data_file)
    return args


def combine_files(src_directory):
    all_data = pd.DataFrame()
    for f in glob.glob(os.path.join(src_directory, "/*.txt")):
        df = pd.read_csv(
            f,
            index_col=None,
            header=None,
            error_bad_lines=False,
            sep=",",
            warn_bad_lines=False,
        )
        df.columns = ["ID", "Status", "Time", "Date", "Call Letters", " ", " "]
        all_data = all_data.append(df, ignore_index=True)
    return all_data


def do_things():

    df = pd.read_csv()
    all_data = pd.merge(df)
    all_data.drop_duplicates(subset=None, inplace=True)
    list_drop = ["ID", "Status", "Time", "Date", " ", " "]
    all_data.drop(list_drop, axis=1, inplace=True)
    all_data.drop_duplicates(subset=None, inplace=True)
    all_data.transpose()
    return all_data


def save_results(all_data):
    all_data.to_csv("Extraction.txt", header=None, index=False)


if __name__ == "__main__":
    parse_args()

    print("Done")
```

When I run this I get an error 
``` usage: call_letter_extract_test.py [-h] data_directory output_directory
call_letter_extract_test.py: error: unrecognized arguments: workspace\call_letter_extractor\Enter Filename
```

**Comment by Sturla22** [comment]
I am having similar issues with MultiFileChooser on Windows.

The only way I could get it to work locally was by changing https://github.com/chriskiehl/Gooey/blob/f771dc66d5173b3800b195d7abf6d9ad9fc57d1a/gooey/gui/formatters.py#L21
to join on `os.pathsep` instead of `' '` and then splitting the paths in the [argparse type](https://docs.python.org/3/library/argparse.html?highlight=argparse#type):
```python
def create_file_list(s: str) -> List[pathlib.Path]:
    ps = [pathlib.Path(x) for x in s.split(os.pathsep)]
    if not all(p.is_file() for p in ps):
        raise ValueError("Not all paths are a file")
    return ps
```
